lib/vmdb/loggers.rb

```diff
@@ -76,6 +80,37 @@ def self.create_logger(log_file_name, logger_class = VMDBLogger)
       nil
     end
 
+    private_class_method def self.progname_from_file(log_file_name)
+      File.basename(log_file_name, ".*")
+    end
+
+    private_class_method def self.create_wrapper_logger(log_file_name, logger_class, wrapped_logger)
+      log_file_name = Pathname.new(log_file_name) unless log_file_name.kind_of?(Pathname)
+      progname      = progname_from_file(log_file_name)
+
+      logger_class.new(nil, progname: progname).tap do |logger|
+        # HACK: In order to access the "filename" of the wrapped logger, we inject it as an instance var.
+        logger.instance_variable_set(:@log_file_name, log_file_name)
+
+        def logger.filename
+          @log_file_name
+        end
+
+        # HACK: In order to access the wrapped logger in test, we inject it as an instance var.
+        if Rails.env.test?
+          logger.instance_variable_set(:@wrapped_logger, wrapped_logger)
+
+          def logger.wrapped_logger
```

Comment:
does it make sense to just prepend this class rather than the 2 method definitions we are throwing on there?

also, if you do decide to prepend the class, always adding the `wrapped_logger` instead of only doing that in test seems good to me to simplify things.

Reply:
No because we only want to change this one instance (though I guess I could create a new "WrappedLogger" class).

> also, if you do decide to prepend the class, always adding the wrapped_logger instead of only doing that in test seems good to me to simplify things.

Are you saying just get rid of the `if Rails.env.test?` part and always make it accessible?  If so, I'd be ok with that.

Reply:
I think I'm in sort of a middle ground.  I don't mind always setting `logger.instance_variable_set(:@wrapped_logger, wrapped_logger)`, but I don't really like the idea of `def logger.wrapped_logger` in any environment (prod/dev/test).

- I like that the test has to `log.instance_variable_get(:@wrapped_logger)` because it really shows that it's testing an internal thing. 
- But, having `def logger.wrapped_logger` makes it feel like this is public API, a thing that you can manipulate and that's not the case.

Reply:
I can do this in a follow up.